tools/devops/device-tests.yml

```diff
@@ -0,0 +1,236 @@
+resources:
+  repositories:
+  - repository: maccore
+    type: github
+    name: xamarin/maccore
+    ref: refs/heads/device-tests-yaml
+    endpoint: xamarin
+
+jobs:
+- job: macOS
+  displayName: xamarin macios
+  timeoutInMinutes: 1000
+
+  pool:
+    name: VSEng-Xamarin-Mac-Devices
+    demands: sh
+
+  steps:
+  - checkout: self
+  - checkout: maccore
+
+  - bash: ./xamarin-macios/tools/devops/system-info.sh
+    displayName: System info
+
+  - bash: cd xamarin-macios && ./configure --enable-xamarin
+    displayName: 'Enable Xamarin'
+
+  - script: |
+     set -x
```

Comment:
Minor, but I've used a two-character indentation in other yml files, a single space looks rather puny: https://github.com/xamarin/xamarin-macios/blob/d5135cce25e60419b70093b825d292765cc2a842/tools/devops/build-samples.yml#L136-L138